SparkyFitness update fails

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

sparkyfitness

### 📂 What was the exact command used to execute the script?

update

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

9.1.5

### 📝 Provide a clear and concise description of the issue.

Currently, SparkyFitness won't update because of a failing fronted build. It seems like an upstream issue, but since we use Node 25 (instead of upstream 20) I'll test with Node 20.

### 🔄 Steps to reproduce the issue.

1. Run `update`
2. See error

### ❌ Paste the full error output (if available).
```bash
   _____                  __         _______ __                      
  / ___/____  ____ ______/ /____  __/ ____(_) /_____  ___  __________
  \__ \/ __ \/ __ `/ ___/ //_/ / / / /_  / / __/ __ \/ _ \/ ___/ ___/
 ___/ / /_/ / /_/ / /  / ,< / /_/ / __/ / / /_/ / / /  __(__  |__  ) 
/____/ .___/\__,_/_/  /_/|_|\__, /_/   /_/\__/_/ /_/\___/____/____/  
    /_/                    /____/                                    

Hit:1 http://security.debian.org trixie-security InRelease
Hit:2 http://deb.debian.org/debian trixie InRelease                       
Hit:3 http://deb.debian.org/debian trixie-updates InRelease               
Hit:4 https://apt.postgresql.org/pub/repos/apt trixie-pgdg InRelease       
Hit:5 https://deb.nodesource.com/node_25.x nodistro InRelease              
All packages are up to date.                         

changed 10 packages in 1s

15 packages are looking for funding
  run `npm fund` for details
  ✔️   Update Node.js 25
  ✔️   Update available: sparkyfitness 0.16.4.6 → 0.16.4.7
  ✔️   Stopped Services
  ✔️   Backed up data
  ✔️   Deployed: sparkyfitness (0.16.4.7)
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated lodash.get@4.4.2: This package is deprecated. Use the optional chaining (?.) operator instead.
npm warn deprecated lodash.isequal@4.5.0: This package is deprecated. Use require('node:util').isDeepStrictEqual instead.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@7.1.6: Glob versions prior to v9 are no longer supported

added 588 packages, and audited 589 packages in 7s

106 packages are looking for funding
  run `npm fund` for details

6 vulnerabilities (1 low, 1 moderate, 3 high, 1 critical)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
  ✔️   Updated Sparky Fitness Backend
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated whatwg-encoding@3.1.1: Use @exodus/bytes instead for a more spec-conformant and faster implementation
npm warn deprecated glob@7.2.3: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me
npm warn deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm warn deprecated source-map@0.8.0-beta.0: The work that was done in this beta branch won't be included in future versions
npm warn deprecated glob@10.5.0: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me
npm warn deprecated glob@10.5.0: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me
npm warn deprecated glob@10.5.0: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me

added 1081 packages, and audited 1082 packages in 12s

211 packages are looking for funding
  run `npm fund` for details

7 vulnerabilities (1 low, 6 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> sparkyfitnessfrontend@0.16.4.7 build
> tsc -b && vite build

src/api/Exercises/exerciseEntryService.ts:254:12 - error TS2304: Cannot find name 'ExerciseProgressData'.

254 ): Promise<ExerciseProgressData[]> => {
               ~~~~~~~~~~~~~~~~~~~~

src/api/Exercises/exerciseEntryService.ts:267:31 - error TS2304: Cannot find name 'ExerciseProgressData'.

267   return response.map((entry: ExerciseProgressData) => ({
                                  ~~~~~~~~~~~~~~~~~~~~

src/pages/Settings/CalculationSettings.tsx:45:38 - error TS2307: Cannot find module '@/hooks/Diary/useDiaryInvalidation' or its corresponding type declarations.

45 import { useDiaryInvalidation } from '@/hooks/Diary/useDiaryInvalidation';
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/pages/Settings/CalculationSettings.tsx:73:35 - error TS2304: Cannot find name 'useDailyProgressInvalidation'.

73   const invalidateDailyProgress = useDailyProgressInvalidation();
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 4 errors.

  ✖️   in line 59: exit code 2 (Misuse of shell builtins (e.g. syntax error)): while executing command  npm run build

  --- Last 20 lines of log ---
[2026-03-01 11:39:34] [INFO] Update Node.js 25
[2026-03-01 11:39:37] [OK] Update Node.js 25
[2026-03-01 11:39:37] [INFO] Checking for update: sparkyfitness
[2026-03-01 11:39:38] [OK] Update available: sparkyfitness 0.16.4.6 → 0.16.4.7
[2026-03-01 11:39:38] [INFO] Stopping Services
[2026-03-01 11:39:38] [OK] Stopped Services
[2026-03-01 11:39:38] [INFO] Backing up data
[2026-03-01 11:39:38] [OK] Backed up data
[2026-03-01 11:39:38] [INFO] Fetching GitHub release: sparkyfitness (0.16.4.7)
[2026-03-01 11:39:42] [OK] Deployed: sparkyfitness (0.16.4.7)
[2026-03-01 11:39:42] [INFO] Updating Sparky Fitness Backend
[2026-03-01 11:39:49] [OK] Updated Sparky Fitness Backend
[2026-03-01 11:39:49] [INFO] Updating Sparky Fitness Frontend (Patience)
[2026-03-01 11:40:19] [ERROR] in line 59: exit code 2 (Misuse of shell builtins (e.g. syntax error)): while executing command  npm run build
  -----------------------------------
```
### 🖼️ Additional context (optional).

_No response_

--- a/ct/sparkyfitness.sh
+++ b/ct/sparkyfitness.sh
@@ -29,7 +29,8 @@ function update_script() {
     exit
   fi
 
-  NODE_VERSION="25" NODE_MODULE="pnpm" setup_nodejs
+  PNPM_VERSION="$(jq -r '.packageManager | split("@")[1]' /opt/sparkyfitness/package.json)"
+  NODE_VERSION="25" NODE_MODULE="pnpm@${PNPM_VERSION}" setup_nodejs
 
   if check_for_gh_release "sparkyfitness" "CodeWithCJ/SparkyFitness"; then
     msg_info "Stopping Services"

--- a/ct/sparkyfitness.sh
+++ b/ct/sparkyfitness.sh
@@ -29,9 +29,6 @@ function update_script() {
     exit
   fi
 
-  PNPM_VERSION="$(jq -r '.packageManager | split("@")[1]' /opt/sparkyfitness/package.json)"
-  NODE_VERSION="25" NODE_MODULE="pnpm@${PNPM_VERSION}" setup_nodejs
-
   if check_for_gh_release "sparkyfitness" "CodeWithCJ/SparkyFitness"; then
     msg_info "Stopping Services"
     systemctl stop sparkyfitness-server nginx
@@ -48,6 +45,9 @@ function update_script() {
     msg_ok "Backed up data"
 
     CLEAN_INSTALL=1 fetch_and_deploy_gh_release "sparkyfitness" "CodeWithCJ/SparkyFitness" "tarball"
+
+    PNPM_VERSION="$(jq -r '.packageManager | split("@")[1]' /opt/sparkyfitness/package.json)"
+    NODE_VERSION="25" NODE_MODULE="pnpm@${PNPM_VERSION}" setup_nodejs
 
     msg_info "Updating Sparky Fitness Backend"
     cd /opt/sparkyfitness/SparkyFitnessServer

--- a/ct/sparkyfitness.sh
+++ b/ct/sparkyfitness.sh
@@ -29,7 +29,7 @@ function update_script() {
     exit
   fi
 
-  NODE_VERSION="25" setup_nodejs
+  NODE_VERSION="25" NODE_MODULE="pnpm" setup_nodejs
 
   if check_for_gh_release "sparkyfitness" "CodeWithCJ/SparkyFitness"; then
     msg_info "Stopping Services"
@@ -55,8 +55,8 @@ function update_script() {
 
     msg_info "Updating Sparky Fitness Frontend (Patience)"
     cd /opt/sparkyfitness/SparkyFitnessFrontend
-    $STD npm install
-    $STD npm run build
+    $STD pnpm install
+    $STD pnpm run build
     cp -a /opt/sparkyfitness/SparkyFitnessFrontend/dist/. /var/www/sparkyfitness/
     msg_ok "Updated Sparky Fitness Frontend"
 

--- a/install/sparkyfitness-install.sh
+++ b/install/sparkyfitness-install.sh
@@ -17,7 +17,7 @@ msg_info "Installing Dependencies"
 $STD apt install -y nginx
 msg_ok "Installed Dependencies"
 
-NODE_VERSION="25" setup_nodejs
+NODE_VERSION="25" NODE_MODULE="pnpm" setup_nodejs
 PG_VERSION="18" setup_postgresql
 PG_DB_NAME="sparkyfitness" PG_DB_USER="sparky" PG_DB_GRANT_SUPERUSER="true" setup_postgresql_db
 
@@ -50,8 +50,8 @@ msg_ok "Built Backend"
 
 msg_info "Building Frontend (Patience)"
 cd /opt/sparkyfitness/SparkyFitnessFrontend
-$STD npm install
-$STD npm run build
+$STD pnpm install
+$STD pnpm run build
 cp -a /opt/sparkyfitness/SparkyFitnessFrontend/dist/. /var/www/sparkyfitness/
 msg_ok "Built Frontend"
 

--- a/install/sparkyfitness-install.sh
+++ b/install/sparkyfitness-install.sh
@@ -17,11 +17,13 @@ msg_info "Installing Dependencies"
 $STD apt install -y nginx
 msg_ok "Installed Dependencies"
 
-NODE_VERSION="25" NODE_MODULE="pnpm" setup_nodejs
 PG_VERSION="18" setup_postgresql
 PG_DB_NAME="sparkyfitness" PG_DB_USER="sparky" PG_DB_GRANT_SUPERUSER="true" setup_postgresql_db
 
 fetch_and_deploy_gh_release sparkyfitness "CodeWithCJ/SparkyFitness" "tarball" "latest"
+
+PNPM_VERSION="$(jq -r '.packageManager | split("@")[1]' /opt/sparkyfitness/package.json)"
+NODE_VERSION="25" NODE_MODULE="pnpm@${PNPM_VERSION}" setup_nodejs
 
 msg_info "Configuring Sparky Fitness"
 mkdir -p "/etc/sparkyfitness" "/var/lib/sparkyfitness/uploads" "/var/lib/sparkyfitness/backup" "/var/www/sparkyfitness"

